docs/source/DeveloperGuide/APIs/index.rst

```diff
@@ -0,0 +1,111 @@
+.. _apis:
+
+Systems APIs
+============
+
+Currently, client-server interaction with DIRAC systems is provided by the HTTP services that writen on the :py:class:`~DIRAC.Core.Tornado.Server.TornadoService` framework.
+This framework describe a fixed interface via HTTP POST requests as PRC calls, see :ref:`httpsTornado`.
+Starting with 8.0, an additional :py:class:`~ DIRAC.Core.Tornado.Server.TornadoREST` framework was created, which expands the possibilities for writing access interfaces to DIRAC systems.
+
+Consider the structure:
+-----------------------
+
+The main change introduced here is the separation in a :py:class:`~ DIRAC.Core.Tornado.Server.private.BaseRequestHandler` class of the basic logic for authorizing the request, search and run the target method in the thread.
+Thus TornadoService and TornadoREST following BaseRequestHandler can describe RPC and REST interfaces respectively::
+
+    BaseRequestHandler
+    |
+    |\_TornadoService
+    |
+     \_TornadoREST
+
+From the scheme it is intuitively clear that on the basis of BaseRequestHandler it is possible to create other framework if flexibility of existing is not enough.
+
+New components:
++++++++++++++++
+
+  - :py:class:`~ DIRAC.Core.Tornado.Server.private.BaseRequestHandler` contains the basic logic described here :ref:`httpsTornado` except for the access interface.
+  - :py:class:`~ DIRAC.Core.Tornado.Server.TornadoREST` framework for writing REST API for DIRAC systems
+
+What is the purpose of this?
+----------------------------
+
+The purpose of this component is to build DIRAC system status access interfaces based on HTTP requests outside the fixed RPC request interface described in TornadoServices.
+
+The main reason for the implementation of this feature is the need to implementation of API for DIRAC Authorization Server based on OAuth2 framework,
+which in turn is an integral part of the implementation of OAuth2 DIRAC authorization.
+So currently, there is only one API implementation for the Framework system, which is :py:class:`~ DIRAC.FrameworkSystem.API.AuthHandler`.
+For the same reason, the ability to authorize with an access token (see https://datatracker.ietf.org/doc/html/rfc6750#section-2) was added to the authorization steps.
+
+How to write APIs
+-----------------
+
+The following is an example of a simple API:
+
+.. literalinclude::
```

Comment:
If you use `.. literalinclude::`  magic you provide a py file: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude

Reply:
fixed